libbeat/esleg/eslegclient/connection_test.go

```diff
@@ -104,9 +111,43 @@ func TestHeaders(t *testing.T) {
 		require.NoError(t, err)
 
 		require.Equal(t, req.Header, http.Header(td.expected))
+
 	}
 }
 
+func TestUserAgentHeader(t *testing.T) {
+	metricsReg := metrics.NewRegistry()
+	version.SetPackageVersion("8.15.0")
+
+	mux := http.NewServeMux()
+	// create a fake elasticsearch endpoint to track the user-agent
+	mockHandler := mock.NewAPIHandler(uuid.New(), "", metricsReg, time.Now().Add(time.Hour), 0, 0, 0, 0, 0)
+	mux.Handle("/", mockHandler)
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
+	conn, err := NewConnection(ConnectionSettings{
+		URL:              server.URL,
+		Beatname:         "testbeat",
+		UserAgentPostfix: "Agent",
+	})
+	require.NoError(t, err)
+
+	err = conn.Connect()
+	require.NoError(t, err)
+
+	// the metrics object passed to the mock handler contains a list of reported user-agents, search those
+	found := false
+	for key := range metricsReg.GetAll() {
+		if strings.Contains(key, "testbeat-Agent") {
```

Comment:
Can this be flaky? Is this guaranteed to be populated once `Connect()` returns above?

Reply:
The request in `Connect()` happens synchronously, I don't think it should be possible for us to end up in a situation where we've completed the `Connect()` but the underlying metrics hashmap in the mock library hasn't updated.